crates/fluvio-spu-schema/src/produce/request.rs

```diff
@@ -41,7 +41,7 @@ pub struct ProduceRequest<R> {
     /// Each topic to produce to.
     pub topics: Vec<TopicProduceData<R>>,
 
-    #[fluvio(min_version = PRODUCER_TRANSFORMATION_API)]
```

Comment:
This is regression.  Lets fix this in this PR

Reply:
I suggest removing this line, which will be equivalent to what we have now. And deal with setting `min_version` to 8 in a separate issue.

Reply:
There are couple of things that need to be considered if both are going to be available. 
Currently the code is using the value of `min_version` for example in couple of places passing it as a `Literal` value to `struct`, doing some checks etc, for example here: 

https://github.com/infinyon/fluvio/blob/b0f846dfd618c3e7ece53918ca87391c217016fa/crates/fluvio-protocol-derive/src/ast/prop.rs#L38C25-L38C25

If this also accepts a `Path`, this means this code cannot work and this logic should change, because basically a `Path` is not a variable so we don't know what it's value is or if it actually has any value, the only thing we have is the name of the "variable". Even if you check the current crate, it's trying to use non existent variable without crashing, just because it doesn't know it's a variable because it's just a `Path`.

A way maybe this can work is if this is treated like an "env" variable and again we just have the Path name which will be the same as passing a literal with `"the_variable_name_as_literal"` ( similar to how `serde` does it for their `searlize_with `funcs for example). So it will only help with centralising the min/max versions for example. And again this is based on the idea that the current code should work where the Literals are used like currently in "validate_versions" for example.

That's what currently is on my mind but not really sure what the right direction would be.

Reply:
Let's not worry about supporting `Path` in this PR

Reply:
Problem is that this may not be addressed soon and we will lose consistent check.   @vrrashkov, Your suggestion using serde like approach seems to be good alternative as below:
```
fn min_transformation_api() -> i16 {
    8
}

   #[fluvio(min_version = "min_transformation_api")]
    pub smartmodules: Vec<SmartModuleInvocation>,
```

Reply:
I mentioned using something like a global "`env`" as a way of doing it, because we will need a value for the `proc-macro `and at this point it's too early for the compiler to know if certain functions/variables exist and we are back at the same problem. The code mentioned above tries to make checks during the proc-macro creation so we cannot just call a `function`, but we can create an `env global `variable or a `map` or something similar that will use the `literal` to get the real value at least that's what I am thinking can work with the current setup.

Reply:
@sehz What you are suggesting (support for constants or functions as value) is not trivial. It is worth a separate issue.

Reply:
Sure.  Lets create issue to ensure this gets addressed soon.

Reply:
Created https://github.com/infinyon/fluvio/issues/3673